ts/components/session/ActionsPanel.tsx

```diff
@@ -102,17 +121,47 @@ const Section = (props: { type: SectionType; avatarPath?: string }) => {
     case SectionType.Moon:
       iconType = SessionIconType.Moon;
       break;
-
+    case SectionType.PathIndicator:
+      iconType = SessionIconType.Circle;
+      break;
     default:
       iconType = SessionIconType.Moon;
   }
 
-  const unreadToShow = type === SectionType.Message ? unreadMessageCount : undefined;
+  // calculate light status.
+  // TODO: Refactor this so this logic is reusable elsewhere.
+
+
+  // TEST:
+  if (type === SectionType.PathIndicator) {
+    const onionState = useSelector((state: StateType) => state.onionPaths);
+
+    let statusColor = Constants.UI.COLORS.DANGER;
+    const isOnline = useNetwork();
+    if (!(onionState && onionState.snodePath) || !isOnline) {
+      return <StatusLight isSelected={isSelected} color={Constants.UI.COLORS.DANGER}></StatusLight>;
+    } else {
+
+      const onionSnodePath = onionState.snodePath;
+      if (onionState && onionSnodePath && onionSnodePath.path.length > 0) {
+        let onionNodeCount = onionSnodePath.path.length;
+        statusColor =
+          onionNodeCount > 2
+            ? Constants.UI.COLORS.GREEN
+            : onionNodeCount > 1
+              ? Constants.UI.COLORS.WARNING
+              : Constants.UI.COLORS.DANGER;
+      }
+    }
+
+    return <StatusLight isSelected={isSelected} color={statusColor}></StatusLight>;
+  }
 
```

Comment:
you can keep this like that, but can you put it in another component in ActionPanel.tsx so we keep this part cleaner